File: app/src/main/java/com/rdev/tryp/SetLocationFragment.java
package com.rdev.tryp;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

public class SetLocationFragment extends Fragment {
    ImageButton location_btn;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.twenty_three, container, false);
        location_btn = v.findViewById(R.id.location_btn);
        location_btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                ((MapActivity) getActivity()).zoomToCurrentLocation();
            }
        });
        return v;
    }
}

File: app/src/main/java/com/rdev/tryp/SetLocationFragment.kt
package com.rdev.tryp

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageButton
import androidx.fragment.app.Fragment

class SetLocationFragment : Fragment() {
    private lateinit var btnLocation: ImageButton

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val v = inflater.inflate(R.layout.twenty_three, container, false)
        btnLocation = v.findViewById(R.id.location_btn)
        btnLocation.setOnClickListener { (activity as ContentActivity).zoomToCurrentLocation() }
        return v
    }

}